dcos/http.py

```diff
@@ -106,8 +117,10 @@ def _request(method,
     :type url: str
     :param is_success: Defines successful status codes for the request
     :type is_success: Function from int to bool
-    :param timeout: request timeout
-    :type timeout: int
+    :param timeout: How many seconds to wait for the server to send data
```

Comment:
It is not clear from this what `True` means.

Reply:
This is quite a complicated parameter now.

What do you think of this:

```python
def _request(
    ...
    timeout_from_config=True,
    custom_timeout=None,
):
    if timeout_from_config and custom_timeout is not None:
        raise Exception()
```

?

Reply:
And perhaps that logic can be done in `request` and this function can be kept basically the same as it was before.

Reply:
Indeed the timeout can accept different kind of values, this comes from the requests package, which can accept None, a float or a tuple. Adding a default to True doesn't impact the usage (eg. this will never be needed to do `request(timeout=True)`, rather this is more of an internal way to specify a default (because the default we want has complex logic).

Having to do `request(timeout_from_config=False, custom_timeout=30)` in order to specify a timeout seems very unnatural to me.

Reply:
This is an implementation detail and so I will leave the final decision on the internal API to you and the other core maintainers of this project. However, I do feel that using `True` on top of the `requests` API is hard to understand.

I'll restate the choice that I would make, but I understand that this is not your preference:

`timeout` is a required field.
Consumers can use a helper to get a value from a configuration, or a default.
That might look like:

```python
timeout = helpers.get_timeout_from_config(config=toml_config)
request(timeout=timeout, ...)
```